app/src/main/java/io/github/wulkanowy/ui/modules/message/preview/MessagePreviewPresenter.kt

```diff
@@ -87,6 +93,66 @@ class MessagePreviewPresenter @Inject constructor(
         } else false
     }
 
+    fun onShare(): Boolean {
+        message?.let {
+            var text = "Temat: ${it.subject}\n" + when (it.sender.isNotEmpty()) {
+                true -> "Od: ${it.sender}\n"
+                false -> "Do: ${it.recipient}\n"
+            } + "Data: ${it.date.toFormattedString("yyyy-MM-dd HH:mm:ss")}\n\n${it.content}"
+
+            attachments?.let {
+                if (it.size > 0) {
+                    text += "\n\nZałączniki:"
+
+                    it.forEach { attachment ->
+                        text += "\n${attachment.filename}: ${attachment.url}"
+                    }
+                }
+            }
+
+            view?.shareText(text, "FW: ${it.subject}")
+            return true
+        }
+        return false
+    }
+
+    fun onPrint(): Boolean {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            message?.let {
+                var infoContent = ""
+                infoContent += "<div>" +
+                    "<h4>Data wysłania</h4>" +
+                    it.date.toFormattedString("yyyy-MM-dd HH:mm:ss") +
+                    "</div>"
+                if (it.sender.isNotEmpty()) {
+                    infoContent += "<div>" +
+                        "<h4>Od</h4>" +
+                        it.sender +
+                        "</div>"
+                } else {
+                    infoContent += "<div>" +
+                        "<h4>Do</h4>" +
+                        it.recipient +
+                        "</div>"
+                }
```

Comment:
To można zrobić w sposób bardziej skondensowany i bez `var`a, np. 

```suggestion
				val date = it.date.toFormattedString("yyyy-MM-dd HH:mm:ss")
				val infoContent = "<div><h4>Data wysłania</h4>${date}</div>" + when {
					it.sender.isNotEmpty() -> "<div><h4>Od</h4>${it.sender}</div>"
					else -> "<div><h4>Do</h4>${it.recipient}</div>"
				}
```